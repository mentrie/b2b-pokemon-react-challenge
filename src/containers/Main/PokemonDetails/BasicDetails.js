import React from 'react';
import PropTypes from 'prop-types';
import backArrowIcon from '../../../static/images/back-arrow.svg';
import PokemonItem from '../Pokemons/PokemonItem';
import { formatName } from '../../../helpers/utils';

const BasicDetails = ({ details, backToHome }) => {
  return (
    <div className="pokemon__details">
      <div>
        <img src={backArrowIcon} onClick={backToHome} alt="back arrow" />
        <span className="pokemon__name">{formatName(details.name)}</span>
      </div>
      <div className="pokemon__basic__details">
        <PokemonItem pokemon={details.forms[0]} />
        <div className="pokemon__details__items">
          <div className="basic__item">
            <span>Name:</span>
            <span>{details.name}</span>
          </div>
          <div className="basic__item">
            <span>Weight:</span>
            <span>{details.weight}</span>
          </div>
          <div className="basic__item">
            <span>{`Type${details.types.length > 1 ? 's' : ''}`}:</span>
            <span>{details.types.map(({type}) => type.name).join(', ')}</span>
          </div>
          <div className="basic__item">
            <span>{`Abilit${details.abilities.length > 1 ? 'ies' : 'y'}`}:</span>
            <span>{details.abilities.map(({ability}) => ability.name).join(', ')}</span>
          </div>
          <div className="basic__item">
            <span>Evolution Info:</span>
            <span>{details.base_experience}</span>
          </div>
        </div>
      </div>
    </div>
  );
};

BasicDetails.propTypes = {
  details: PropTypes.shape({
    name: PropTypes.string,
    types: PropTypes.array,
    abilities: PropTypes.array,
    base_experience: PropTypes.number
  }),
  backToHome: PropTypes.func.isRequired
};

export default BasicDetails;
