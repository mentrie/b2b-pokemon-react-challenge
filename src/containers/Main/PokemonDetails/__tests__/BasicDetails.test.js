import React from 'react';
import BasicDetails from '../BasicDetails';

let props;
beforeAll(() => {
  props = {
    backToHome: jest.fn(),
    details: {
      name: '',
      types: [{type: 'test'}],
      abilities: [{ability: 'test'}],
      forms: [{name: '', url: 'test/1'}]
    }
  };
});
test('render component', () => {
  const component = shallow(<BasicDetails {...props} />);
  expect(component).toMatchSnapshot();
});

test('handles returning back to homepage', () => {
  const component = mount(<BasicDetails {...props} />);
  component.find('img').first().simulate('click');
  expect(props.backToHome).toHaveBeenCalled();
});
