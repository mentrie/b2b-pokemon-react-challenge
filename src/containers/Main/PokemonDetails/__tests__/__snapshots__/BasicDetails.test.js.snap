// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component 1`] = `
<div
  className="pokemon__details"
>
  <div>
    <img
      alt="back arrow"
      onClick={[Function]}
      src="back-arrow.svg"
    />
    <span
      className="pokemon__name"
    />
  </div>
  <div
    className="pokemon__basic__details"
  >
    <PokemonItem
      pokemon={
        Object {
          "name": "",
          "url": "test/1",
        }
      }
    />
    <div
      className="pokemon__details__items"
    >
      <div
        className="basic__item"
      >
        <span>
          Name:
        </span>
        <span />
      </div>
      <div
        className="basic__item"
      >
        <span>
          Weight:
        </span>
        <span />
      </div>
      <div
        className="basic__item"
      >
        <span>
          Type
          :
        </span>
        <span />
      </div>
      <div
        className="basic__item"
      >
        <span>
          Ability
          :
        </span>
        <span />
      </div>
      <div
        className="basic__item"
      >
        <span>
          Evolution Info:
        </span>
        <span />
      </div>
    </div>
  </div>
</div>
`;
